### PR TITLE
[p2] Added event types, stricter static constants, fixed getBodyByID, added missing constructor options, prettier formatted.

### PR DESCRIPTION
**Pull request by jramstedt**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/schteppe/p2.js
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Review by peterblazejewicz — CHANGES_REQUESTED** [review]
Why do you want to change handle to at `alexandermbrown`? if this legit, one needs to explicit mention that intent in the PR and there should be ACK from this person given

File: types/p2/index.d.ts [hunk 1]
```diff
@@ -27,27 +24,25 @@ declare namespace p2 {
         overlaps(aabb: AABB): boolean;
         containsPoint(point: [number, number]): boolean;
         overlapsRay(ray: Ray): number;
-
     }
 
     export class Broadphase {
+        static AABB: 1;
+        static BOUNDING_CIRCLE: 2;
```
Comment (peterblazejewicz) [inline types/p2/index.d.ts:31]: are those constants or static properties (of type `number`)? The source uses those as props, but the intent could be to treat those as constants

**Comment by alexandermbrown** [comment]
Could someone please remove my name and the old JellyAlex from the 'Definitions by' list? Sorry for the hassle, I didn't really know what it meant when I initially contributed.

File: types/p2/index.d.ts (hunk 1 above)
Comment (jramstedt) [reply types/p2/index.d.ts:31]: These are static members in the class.
'readonly' is for instance fields.
So I can't change them to 'readonly'.
There are no readonly or constant statics in TypeScript.

**Comment by peterblazejewicz** [comment]
> Could someone please remove my name and the old JellyAlex from the 'Definitions by' list? Sorry for the hassle, I didn't really know what it meant when I initially contributed.

Here #54993

File: types/p2/index.d.ts (hunk 1 above)
Comment (peterblazejewicz) [reply types/p2/index.d.ts:31]: that's perfectly possible:
```ts
declare class MyClass {
    static readonly myField: number;
}
```
but no worries, that was just question about current implementation and actual usage.

File: types/p2/index.d.ts [hunk 2]
```diff
@@ -1,21 +1,18 @@
 // Type definitions for p2.js v0.7.1
 // Project: https://github.com/schteppe/p2.js/
-// Definitions by: Clark Stevenson <https://github.com/clark-stevenson>, Janne Ramstedt <https://github.com/jramstedt>, Alex Brown <https://github.com/JellyAlex>
+// Definitions by: Clark Stevenson <https://github.com/clark-stevenson>, Janne Ramstedt <https://github.com/jramstedt>, Alex Brown <https://github.com/alexandermbrown>
```
Comment (peterblazejewicz) [inline types/p2/index.d.ts:3]:
```suggestion
// Definitions by: Clark Stevenson <https://github.com/clark-stevenson>, Janne Ramstedt <https://github.com/jramstedt>, Alex Brown <https://github.com/JellyAlex>
```

Comment (peterblazejewicz) [reply types/p2/index.d.ts:3]: Please accept this suggestion, I've created separate PR to remove Alex from code owners (as requested in a comment)

**Review by MRck23 — APPROVED** [review]

**Review by peterblazejewicz — APPROVED** [review]

**Comment by jramstedt** [comment]
Ready to merge

**Pull request merged**